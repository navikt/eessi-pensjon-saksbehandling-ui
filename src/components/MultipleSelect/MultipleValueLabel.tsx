import { components } from 'react-select'
import styled from 'styled-components'

const Label = styled.div`
  padding: 0px;
`
const FlexDiv = styled.div`
  display: flex;
  align-items: center;
`
const MultipleValueLabel = (props: any) => (
  <components.MultiValueLabel data-test-id='c-multipleselect-multivaluelabel' {...props}>
    <FlexDiv>
      <Label>&nbsp;&nbsp;&nbsp;&nbsp;{props.data.label}&nbsp;&nbsp;&nbsp;&nbsp;</Label>
    </FlexDiv>
  </components.MultiValueLabel>
)

export default MultipleValueLabel
