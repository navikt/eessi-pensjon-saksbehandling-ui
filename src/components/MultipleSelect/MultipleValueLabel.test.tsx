
import MultipleValueLabel from 'components/MultipleSelect/MultipleValueLabel'
import { render } from '@testing-library/react'

describe('components/MultipleSelect/MultipleValueLabel', () => {
  let wrapper: any
  const initialMockProps = {
    data: {
      label: 'mockLabel'
    }
  }

  beforeEach(() => {
    // @ts-ignore
    wrapper = render(<MultipleValueLabel {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Render: match snapshot', () => {
    const { container } = render(<MultipleValueLabel {...initialMockProps} />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('Render: has proper HTML structure', () => {
    expect(wrapper.find('[data-testid=\'c-multipleselect-multivaluelabel\']')).toBeTruthy()
    expect(wrapper.find('[data-testid=\'c-multipleselect-multivaluelabel\']').render().text().trim()).toEqual(
      initialMockProps.data.label
    )
  })
})
