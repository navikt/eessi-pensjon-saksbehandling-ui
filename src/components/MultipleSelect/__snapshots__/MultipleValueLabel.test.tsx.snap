// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MultipleSelect/MultipleValueLabel Render: match snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

ui-fss.g1[id="c0"] {
  content: "c1,";
}

<MultipleValueLabel
  data={
    Object {
      "label": "mockLabel",
    }
  }
>
  <MultiValueGeneric
    data={
      Object {
        "label": "mockLabel",
      }
    }
    data-testid="c-multipleselect-multivaluelabel"
  >
    <div>
      <styled.div>
        <div
          className="c0 c1"
        >
          <ForwardRef>
            <p
              className="navds-body-long"
            >
              mockLabel
            </p>
          </ForwardRef>
        </div>
      </styled.div>
    </div>
  </MultiValueGeneric>
</MultipleValueLabel>
`;
